src/main/java/org/llorllale/cactoos/matchers/MatchesRegex.java

```diff
@@ -50,12 +52,17 @@ public MatchesRegex(final String regex) {
      */
     public MatchesRegex(final Text regex) {
         super(
-            new TextMatcher(
-                new MatcherOf<>(
-                    (String act) -> act.matches(regex.asString()),
-                    regex
+            new MatcherOf<>(
+                act -> act.asString().matches(regex.asString()),
+                desc -> desc.appendText(
+                    new FormattedText(
+                    "Text matches /%s/",
+                        Pattern.compile(regex.asString()).toString()
+                    ).asString()
                 ),
-                "Text matches "
+                (actual, desc) -> desc.appendText(
+                    new FormattedText("Text is \"%s\"", actual).asString()
+                )
```

Comment:
@baudoliver7 why not reuse `TextMatcher` here? at least we can pass the `MatcherOf` to it and rely on `TextMatcher` predefined `actual` for example.

Reply:
@victornoel It's because we used `/` instead of `"` as delimiter of regex in description text formatting. If we had used the old format we would have the following code :
```java
public MatchesRegex(final Text regex) {
  super(
      new TextMatcher(
          regex,
          (act, txt) -> act.matches(txt),
          "Text matches"
      )
  );
}
```
Isn't it more convenient ?

Reply:
@baudoliver7 yes, you are right, let's do like you propose in your last comment for now, sorry about the back and forth :)

Reply:
@victornoel Done.